Compiler-Internals.md

```diff
@@ -18,19 +19,32 @@ The checker then waits until a particular symbol is requested to calculate type
 Each phase of the compiler (parser, binder, etc -- see below for details) treats data structures from the previous phases as immutable.
 In addition, data structures created within each phase are not usually modified after their creation.
 This requires a look-aside table in some cases.
-For example, because the binder only looks at one file at a time, the checker needs a merged-symbols table to track merged declarations. 
+For example, because the binder only looks at one file at a time, the checker needs a merged-symbols table to track merged declarations.
 It checks whether a symbol has an entry in the merged-symbols table each time before it uses a symbol.
 
 # Parser
 
+It's a recursive descent parser.
+It's pretty resilient to small changes, so if you search for function names matching the thing you want to change, you can probably get away with not having to think about the whole parser.
+There aren't any surprises in the general implementation style here.
+
 ## Incremental parsing
 
 ## ECMAScript parsing contexts
 
 ## ECMAScript automatic semicolon insertion
 
+## JSDoc parsing
+
+# Binder
+
 # Checker
 
+The checker is almost 20,000 lines long, and does almost everything that's not syntactic.
```

Comment:
Just mention that it's one of two semantic passes, and that it's the primary one.